### ISSUE TITLE
False-Positive when detecting rails application

### ISSUE THREAD

**Issue by Bartuz** [zendesk/racecar#119]
TL;DR
`require 'rails'` returns true for non-rails project with "activesuppport" or "railties"
----

Detection if the app is a rails app happens [here](https://github.com/zendesk/racecar/blob/master/lib/racecar/rails_config_file_loader.rb#L6-L9)

which produces false positive when app uses activesupport. Rails object is then defined, but it has no "logger" object so [these lines](https://github.com/zendesk/racecar/blob/master/lib/racecar/rails_config_file_loader.rb#L23-L28) raise an exception (undefined method on nil object :) )

It was detected in a closed-source project, I can share part of my Gemfile.lock

```
$ cat Gemfile.lock G rail
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    rails-dom-testing (2.0.3)
    rails-html-sanitizer (1.0.4)
    railties (5.1.6)
      railties (>= 4.2.7, < 5.3.0)
```

and

```
$ cat Gemfile.lock G active
      activesupport (= 5.1.6)
      activesupport (= 5.1.6)
    activemodel (5.1.6)
      activesupport (= 5.1.6)
    activerecord (5.1.6)
      activemodel (= 5.1.6)
      activesupport (= 5.1.6)
    activesupport (5.1.6)
      activesupport (>= 4.2.0)
      activesupport (= 5.1.6)
      activerecord (>= 4.2.7, < 5.3.0)
  activerecord (~> 5.1)
  activesupport (~> 5.1)
```

**Comment by Bartuz** [comment]
Our temporary fix which apparently does not break anything


```
#kafka_setup.rb

Racecar::RailsConfigFileLoader.class_eval do
  def self.load!
    $stderr.puts "=> Skipping rails detection as monkeypatch in kafka_setup.rb..."
  end
end
```

and include this file in `bundle exec racecar --require kafka_setup.rb`

**Comment by dasch** [comment]
Can you figure out a better way of detecting Rails?

**Comment by Bartuz** [comment]
In our case (our case = no rails but activesupport/railties) script fails on (Rails.logger.extend)[https://github.com/zendesk/racecar/blob/master/lib/racecar/rails_config_file_loader.rb#L27] so kind of ugly hot-fix would be changing
```
   begin
        require "rails"
      rescue LoadError
        # Not a Rails application.
      end
```
to 
```
   class OnlyActiveSupportDetected < LoadError; end

   begin
        require "rails"
        raise OnlyActiveSupportDetected if Rails.logger.blank?
      rescue LoadError, OnlyActiveSupportDetected
        # Not a Rails application.
    end
```

but it's ugly and I don't like it and it can still produce False-Positives (less than before of course, so that's good) but there could be more false-negatives (i.e. Rails apps that have no .logger assigned? Is it possible? I don't know).

**Comment by simonireilly** [comment]
Plus one for a fix on this - using racecar with ActiveRecord and no rails

**Comment by dasch** [comment]
How about a `--no-rails` flag that disables Rails booting?

**Comment by mensfeld** [comment]
Lets share some friendly knowledge and advise :) why not like that? https://github.com/karafka/karafka/blob/master/lib/karafka/cli/install.rb#L31

**Comment by dasch** [comment]
If it works! :heart:

Anyone up for doing a PR?

**Comment by mensfeld** [comment]
Works really well for "us". Also allows to determine that you have rails in the stack (I don't expect anyone sane to include it and not use it at all) prior to being loaded.
